packages/syntax-core/src/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,181 @@
+import { screen, render } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { expect, vi } from "vitest";
+import Modal from "./Modal";
+
+describe("modal", () => {
+  it("should render successfully", () => {
+    const { baseElement } = render(
+      <Modal
+        header="title"
+        onDismiss={() => {
+          /* empty */
+        }}
+      >
+        <p>text</p>
+      </Modal>,
+    );
+    expect(baseElement).toBeTruthy();
+  });
+
+  it("should render children successfully", async () => {
+    render(
+      <Modal
+        header="title"
+        onDismiss={() => {
+          /* empty */
+        }}
+      >
+        <p>text</p>
+      </Modal>,
+    );
+    const text = await screen.findAllByText("text");
+    expect(text).toHaveLength(1);
+  });
+
+  it("should fire onDismiss when X is clicked", async () => {
+    const handleOnDismiss = vi.fn();
+
+    const { baseElement } = render(
+      <Modal header="title" onDismiss={handleOnDismiss}>
+        <p>text</p>
+      </Modal>,
+    );
+
+    // disabling because closeButton can't be accessed by react-testing-lib methods
+    // eslint-disable-next-line testing-library/no-node-access
+    const closeButton = baseElement.querySelector("button") as Element;
+    // eslint-disable-next-line testing-library/no-await-sync-events
+    await userEvent.click(closeButton);
+    expect(handleOnDismiss).toHaveBeenCalledOnce();
+  });
+
+  it("should render header text successfully", async () => {
+    render(
+      <Modal
+        header="title"
+        onDismiss={() => {
+          /* empty */
+        }}
+      >
+        <p>text</p>
+      </Modal>,
+    );
+    const header = await screen.findAllByText("title");
```

Comment:
this one doesn't seem like it can have a `data-testid`, because the text is passed into `heading` component. unless we want to add it to the component? (which i dont think we want to do)

Reply:
@somethiiing makes sense, instead can we use `getByRole` - it also enforces that we use the correct heading level.

```tsx
expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
  "title",
);
```